x/txfees/keeper/fees.go

```diff
@@ -152,19 +150,27 @@ func (k Keeper) swapFeeToBaseDenom(
 	}
 
 	// Swap the coin to base denom
-	var (
-		tokenOutAmount = sdk.ZeroInt() // Token amount in base denom
-		route          = []poolmanagertypes.SwapAmountInRoute{{
-			PoolId:        feetoken.PoolID,
-			TokenOutDenom: baseDenom,
-		}}
-	)
-	err = osmoutils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
-		tokenOutAmount, err = k.poolManager.RouteExactAmountIn(ctx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
-		return err
-	})
+	route := []poolmanagertypes.SwapAmountInRoute{{
+		PoolId:        feetoken.PoolID,
+		TokenOutDenom: baseDenom,
+	}}
+
+	cacheCtx, write := osmoutils.NoEventCacheContext(ctx)
+	tokenOutAmount, err := k.poolManager.RouteExactAmountIn(cacheCtx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
 	if err != nil {
-		return nil, sdk.Coins{takerFeeCoin}, nil
+		return nil, nil, fmt.Errorf("swap fee token: %w", err)
+	}
+	write()
```

Comment:
events not emitted on this `write()` call

they will be manually emitted if we manage to modify the swap event